### PR TITLE
Disable `hostperfcheck` by default

### PR DESCRIPTION
**Pull request by bkchr**
This feature should only be activated by the polkadot binary. Otherwise parachains may accidentally activate this feature.

Closes: https://github.com/paritytech/polkadot/issues/6631

### CODE REVIEW

**Comment by bkchr** [comment]
Also does some other clean ups for when all `*-native` features are deactivated.

**Review by slumber — APPROVED** [review]

File: cli/src/command.rs [hunk 1]
```diff
@@ -238,29 +239,32 @@ macro_rules! unwrap_client {
...
-		#[cfg(not(feature = "hostperfcheck"))]
-		{
-			return Err(PerfCheckError::FeatureNotEnabled { feature: "hostperfcheck" }.into())
-		}
-		#[cfg(feature = "hostperfcheck")]
-		{
-			crate::host_perf_check::host_perf_check()?;
-			return Ok(())
-		}
+		crate::host_perf_check::host_perf_check()?;
```
Comment (mrcnski) [inline cli/src/command.rs:266]: Can we just put `#[cfg(all(feature = "hostperfcheck", build_type = "release"))]` on this function as well as on `Subcommand::HostPerfCheck`? Why have runtime checks that just return `FeatureNotEnabled`?

File: node/service/src/lib.rs [hunk 2]
```diff
@@ -1355,32 +1355,36 @@ pub fn new_chain_ops(
...
 	{
-		return chain_ops!(config, jaeger_agent, telemetry_worker_handle; polkadot_runtime, PolkadotExecutorDispatch, Polkadot)
+		return chain_ops!(config, jaeger_agent, None; polkadot_runtime, PolkadotExecutorDispatch, Polkadot)
 	}
+
 	#[cfg(not(feature = "polkadot-native"))]
-	Err(Error::NoRuntime)
+	{
+		let _ = config;
+		let _ = jaeger_agent;
```
[the remark this answers is not in the document]
Comment (bkchr) [reply node/service/src/lib.rs:1384]: When you don't have any of the `native-*` features enabled, it complains about unused variables. This syntax ensures that they are being "used".

File: cli/src/command.rs (hunk 1 above)
Comment (bkchr) [reply cli/src/command.rs:266]: Sorry, I don't get the question?

Comment (mrcnski) [reply cli/src/command.rs:266]: I mean, why have this `FeatureNotEnabled` error returned at runtime, when we can just disable these code paths with conditional compilation? Then the error is not needed at all, as entering the code path is impossible.

Comment (slumber) [reply cli/src/command.rs:266]: When executing it without the feature compiled you get a help message on what to do in order to run it.
With your suggestion a user gets something like "unknown command"

**Review by mrcnski — APPROVED** [review]

**Pull request merged**